File: greedy/jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
   let n  = nums.length;
   for(let i =0;i<n;i++){
    if(i + nums[i] === n-1) return true;
   } 
   return false;
};

File: greedy/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();

        int prevmax  = nums[0];

        for(int i =0;i<n;i++){

            if(prevmax <0) return false;
            prevmax = max(prevmax,nums[i]);

            if(i+prevmax >= n-1){
                return true;
            }

            prevmax--;
        }
        return false;
    }
};